### PR TITLE
Add support for sync pause/resume

### LINKED ISSUE #2014: Add support for sync pause/resume

### PR DESCRIPTION
**Pull request by emanuelez**
## What, How & Why?

This closes #2014 

## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* [x] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

**Comment by bmunkholm** [comment]
Should it be start/stop or pause/resume as in Cocoa?

**Comment by emanuelez** [comment]
It's start/stop in Java

File: tests/js/session-tests.js [hunk 1]
```diff
@@ -1041,6 +1051,114 @@ module.exports = {
...
+                    }
+                };
+
+                const checks = {
+                    started: false,
+                    stopped: false,
+                    restarted: false,
+                }
+
+                Realm.open(config).then(realm => {
```
Comment (nirinchev) [inline tests/js/session-tests.js]: You can move the `Realm.open` outside the promise constructor. Something like:

```js
Realm.Sync.User.register(...)
    .then((user) => {
        const config = { ... };
        return Realm.open(config);
    })
    .then((realm) => {
        return new Promise(...);
    });
```

File: tests/js/session-tests.js [hunk 2]
```diff
@@ -1041,6 +1051,114 @@ module.exports = {
...
+                    const session = realm.syncSession;
+
+                    session.addConnectionNotification((newState, oldState) => {
+                        if (newState === Realm.Sync.ConnectionState.Connected && checks.started === false) { checks.started = true; session.stop(); }
```
Comment (nirinchev) [inline tests/js/session-tests.js]: This is very convoluted to read. Any reason not to spread them on multiple lines?

File: tests/js/session-tests.js [hunk 3]
```diff
@@ -1041,6 +1051,114 @@ module.exports = {
...
+            return new Promise((resolve, reject) => {
+                let config = {
+                    sync: {
+                        user: u,
+                        url: `realm://localhost:9080/~/${uuid()}`,
+                        fullSynchronization: true,
+                    }
+                };
+
+                Realm.open(config).then(realm => {
```
Comment (nirinchev) [inline tests/js/session-tests.js]: Same here - `Realm.open` can be moved outside the promise constructor

File: tests/js/session-tests.js [hunk 4]
```diff
@@ -1041,6 +1051,114 @@ module.exports = {
...
+    },
+
+    testMultipleStarts() {
+        if(!isNodeProccess) {
+            return;
+        }
+
+        return Realm.Sync.User.register('http://localhost:9080', uuid(), 'password').then((u) => {
+            return new Promise((resolve, reject) => {
+                let config = {
```
Comment (nirinchev) [inline tests/js/session-tests.js]: Should this be `const`?

File: tests/js/session-tests.js (hunk 1 above)
Comment (emanuelez) [reply tests/js/session-tests.js]: I see all the other tests in the file are structured as mine. Also, I cannot concatenate a `.catch(error => reject(error));` at the end of the chain you propose.

Comment (nirinchev) [reply tests/js/session-tests.js]: You don't need to concatenate it because the promise will be rejected anyway.

Comment (nirinchev) [reply tests/js/session-tests.js]: With the current code, you reject a promise if `Realm.open` fails. If you return it as I suggest, you'll end up with the same logical flow without having a codepath for promise rejection.

**Pull request merged**